### PR TITLE
Finally merging in the long-ignored `develop` branch

### PR DESCRIPTION
**Pull request by chrismetcalf**

### CODE REVIEW

**Comment by chrismetcalf** [comment]
@allejo Any chance you could do me a solid and help review this PR? It contains some larger changes and I could use your more experienced eye. Thanks!

**Comment by chrismetcalf** [comment]
@metalmini FYI

**Comment by allejo** [comment]
@chrismetcalf sure thing! I'll take a look later today

File: src/Client.php [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+        return $handle;
+    }
+
+    /**
+     * Execute a CURL resource and json_decode the result
+     *
+     * @param  resource $handle
+     * @return mixed
+     * @throws RuntimeException if the request was not successful
```
Comment (allejo) [inline src/Client.php:168]: `RuntimeException` is no longer thrown here

**Comment by chrismetcalf** [comment]
@allejo @metalmini Got the test suite working under PHP 7.0. Is it worth trying to get it to pass for older versions of PHP, or should I just not look back? Not a PHP expert here.

File: src/Client.php [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+     */
+    private function executeAndDecode($handle)
+    {
+        $response = curl_exec($handle);
```
Comment (allejo) [inline src/Client.php:172]: `curl_exec` may return false if the request fails; might be nice to check for false before feeding it directly to `json_decode`.

File: src/Client.php [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        }
+
+        return $full_url;
+    }
+
+    /**
+     * create cURL handle, which can then be submitted via get
+     *
+     * @param  string $path
+     * @param  array  $params
```
Comment (allejo) [inline src/Client.php:137]: Might be helpful to update these for IDEs to use

File: composer.json [hunk 4]
```diff
@@ -1,13 +1,28 @@
...
     "name": "socrata/soda-php",
     "description": "A simple library to make it easier to access SODA services from PHP",
-    "keywords": ["SODA", "socrata"],
+    "keywords": ["SODA", "socrata", "open data", "sdk"],
     "homepage": "http://dev.socrata.com",
+    "license": "Apache-2.0",
     "require": {
-        "php": "~5.3",
-        "ext-curl": "*"
+        "php": ">=5.3",
```
Comment (allejo) [inline composer.json:8]: Per Socrata's requirement of a minimum TLS, this should be `>=5.6`

**Comment by allejo** [comment]
Regarding older versions of PHP, your call. 5.6 is still supported per the minimum TLS requirement so your call on whether or not you want to test it. PHPUnit 5.x isn't namespaced while 6.x is, so if you do want to test 5.6, then you'd have to bring PHPUnit down a version.

**Pull request closed without merging**